### ISSUE TITLE
"`hub` is not configured. To do it, run `cd $(brew --repository) && hub issue`. Your Github password will be required, but is never stored."

### ISSUE THREAD

**Issue by stvhwrd** [vitorgalvao/tiny-scripts#125]
### TL;DR: Make sure you have defined`GITHUB_USER` and `GITHUB_TOKEN` environment variables.

I just ran into this problem and found a solution.  Leaving this here for future user reference:

#### 1. After installing cask-repair via `brew install vitorgalvao/tiny-scripts/cask-repair`, I attempt to run `cask-repair` for the first time:

**input:**
```
cask-repair --help`
```

**output:**
Usage information is displayed, as expected.

<br>

#### 2. I then attempt to initiate an update of the `googleappengine` cask, and I'm faced with an error.

**input:**
```
cask-repair googleappengine
```

**output:**
```
`hub` is not configured.
To do it, run `cd $(brew --repository) && hub issue`. Your Github password will be required, but is never stored.
```

<br>

#### 3. Puzzling, since I've had [`hub`](https://github.com/github/hub) installed and used frequently for over a year without ever running into this problem.  I do as instructed (several times):

**input:**
```
cd $(brew --repository) && hub issue
```

**output:**
```
   4022] Tell curl to not use .curlrc ( https://github.com/Homebrew/brew/pull/4022 )
   4012] [WIP] cli_parser: Common switch options to be accessible across brew ( https://github.com/Homebrew/brew/pull/4012 )
   4011] brew create: fix meson template ( https://github.com/Homebrew/brew/pull/4011 )
   4007] official_taps: deprecate homebrew/php. ( https://github.com/Homebrew/brew/pull/4007 )
...
```
(etc.)

<br>

#### 4. After reading through the [cask-repair](https://github.com/vitorgalvao/tiny-scripts/blob/master/cask-repair) source, I noticed that it relies on a couple of environment variables - `GITHUB_TOKEN` and `GITHUB_USER`.  To test whether I had defined these, I run a quick `echo`, and this revealed that the `GITHUB_USER` environment variable was not defined.

**input:**
```
echo $GITHUB_USER, $GITHUB_TOKEN
```

**output:**
```
, 23489123jhasdf0923jdfahl2341233rasf89234a912
```
> Note: this is not a valid token - it's just there for illustration

<br>

#### 5.  I added the missing variable to my `~/.zshenv` file (I use zsh and [Prezto](https://github.com/sorin-ionescu/prezto) - you can use `~/.profile` to be safe), sourced it, and re-ran the test from step 4 - this time successful.

**input:**
```
echo export GITHUB_USER="stvhwrd" >> ~/.zshenv && source ~/.zshenv;
echo $GITHUB_USER, $GITHUB_TOKEN
```

**output:**
```
stvhwrd, 23489123jhasdf0923jdfahl2341233rasf89234a912
```

<br>

#### 6.  I tried running cask-repair again - this time successful.

**input:**
```
cask-repair googleappengine
```

**output:**
Taps are updated and cask-repair displays `googleappengine` information, requesting new version as expected.
Problem solved. ✔️

<br> 

\- fin -

**Comment by vitorgalvao** [comment]
> TL;DR: Make sure you have definedGITHUB_USER and GITHUB_TOKEN environment variables.

That is incorrect. You do not need to set those variables. I don’t and never did, and never had an issue. I also don’t remember seeing this problem before. No idea why this happened, but it’s just as likely it was an oddity in your setup. Are you sure you’re on the latest version?

Still, it’s solved for you, so closing the issue.

Thank you anyway for the report, I’ll keep an eye open to see if it happens to anyone else.

**Comment by stvhwrd** [comment]
I cloned the repository an hour before writing this, so yeah definitely the latest version.

```
/usr/local/Cellar/cask-repair/0.36.3 (5 files, 26KB) *
  Built from source on 2018-04-05 at 23:21:19
From: https://github.com/vitorgalvao/homebrew-tiny-scripts/blob/master/cask-repair.rb
```

I guess it's more of a hack than a fix, but maybe it'll help someone in a pinch - that was my intent.   Thanks for your work @vitorgalvao 🍻

**Comment by vvanpo** [comment]
@vitorgalvao I get the exact same result. `hub` works fine, but `cask-repair` doesn't seem to think so.

**Comment by JayTheMarketer** [comment]
@vitorgalvao — happened to me too. did the manually set $GITHUB_USER and now it works.

**Comment by vitorgalvao** [comment]
I’m no longer fixing `hub`-related bugs (Though I will review short PRs to fix issues with it). Next time I touch that, it will be to remove the tool entirely, and likely use the GitHub API directly.

**Comment by JayTheMarketer** [comment]
@vitorgalvao — I only mentioned because you said above that you were going to wait and see if it happened to others. Just wanted to let you know. Thanks for all your work!!

**Comment by vitorgalvao** [comment]
@JayTheMarketer Thank you. It’s just that my goal changed since then, as my frustration with `hub` rose.
